modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CProductLocalServiceImpl.java

```diff
@@ -75,12 +75,10 @@ public CProduct addCProduct(
 	@Override
 	@SystemEvent(type = SystemEventConstants.TYPE_DELETE)
 	public CProduct deleteCProduct(CProduct cProduct) throws PortalException {
-		cProduct = cProductPersistence.remove(cProduct);
-
 		_cpDefinitionLinkLocalService.deleteCPDefinitionLinksByCProductId(
 			cProduct.getCProductId());
 
-		return cProduct;
+		return cProductPersistence.remove(cProduct);
```

Comment:
I'm not sure Brian will merge this

Reply:
@andrea-ale-sbarra , this one still needs review from @alessiorendina  right? I am going to reassign it back to you.

Reply:
yes I would like @alessiorendina to have a look at this if he didn't already.

Thank you!

Reply:
Yeah, I think Brian won't merge this. Let's investigate why this is happening only here, I will reassign this back to @victorcdp

Reply:
@andrea-ale-sbarra @alessiorendina I changed the order because if you remove the product first, when removing the associations, an exception is thrown in [this line](https://github.com/victorcdp/liferay-portal/blob/096535b18478e416716dd67fa9d92ee556725f0f/modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLinkLocalServiceImpl.java#L197), because the product doesn't exist anymore and it can't be found. Do you have any suggestions?

Reply:
I changed the code. Since the problem was the that the previous code was trying to reindex the already deleted product, now the reindex will only trigger if the product hasn't been deleted. I switched from find to fetch since find will thrown an exception if the product isn't found, while fetch just returns null. Can you check if it's acceptable now? @andrea-ale-sbarra